content/codespaces/getting-started-with-codespaces/getting-started-with-your-python-project-in-codespaces.md

```diff
@@ -0,0 +1,221 @@
+# RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+#     && apt-get -y install --no-install-recommends <your-package-list-here>
+
+# [Optional] Uncomment this line to install global node packages.
+# RUN su vscode -c "source /usr/local/share/nvm/nvm.sh && npm install -g <your-package-here>" 2>&1
+```
+
+You can use the Dockerfile to add additional container layers to specify OS packages, node versions, or global packages we want included in our Dockerfile.
+
+### Step 3: Modify your devcontainer.json file 
+
+With your dev container added and a basic understanding of what everything does, you can now make changes to configure it for your environment. In this example, you'll add properties to install npm when your codespace launches and make a list of ports inside the container available locally.
+
+1. In the Explorer, select the `devcontainer.json` file from the tree to open it. You might have to expand the `.devcontainer` folder to see it. 
+
+  !["Codespaces: Rebuild Container" in the command palette](/assets/images/help/codespaces/devcontainers-options.png)  
+
+2. Update your `devcontainer.json` file after `extensions` to include the following:
+
+  ```json{:copy}
+  "forwardPorts": [5000],
+  "postCreateCommand": "pip3 install --user -r requirements.txt",
+  ```
+
+  For more information on `devcontainer.json` properties, see the [devcontainer.json reference](https://code.visualstudio.com/docs/remote/devcontainerjson-reference) on the Visual Studio Code docs.
+
+3. To rebuild your container, access the command palette (`shift command P` / `shift control P`), then start typing "rebuild". Click **Codespaces: Rebuild Container**. 
+
+  ![Rebuild container option](/assets/images/help/codespaces/codespaces-rebuild.png)
+
+  Rebuilding inside your codespace ensures your changes work as expected before you commit the changes to the repository. If something does result in a failure, you’ll be placed in a codespace with a recovery container that you can rebuild from to keep adjusting your container.
+
+
+### Step 4: Run your application
+
+In the previous section, you used the `postCreateCommand` to installing a set of packages via pip3. With our dependencies now installed, we can run our application.
+
+1. Run your start command in the terminal with `python -m flask run`.
+
+2. When your project starts, you should see a toast in the bottom right corner with a prompt to connect to the port your project uses. 
+
+  ![Port forwarding toast](/assets/images/help/codespaces/codespaces-port-toast.png)
+
+### Step 5: Commit your changes
+
+Once you've made changes to your codespace, either new code or configuration changes, you'll want to commit your changes. Committing changes to your repository ensures that anyone else who creates a codespace from this repository has the same configuration. This also means that any customization you do, such as adding {% data variables.product.prodname_vscode %} extensions, will appear for all users.
```

Comment:
From my own experience it's not overly clear that you still need to push to GitHub to save your changes back to the repository. It might be worth calling that out.

Reply:
This is a great point! I'm also writing a new doc that covers source control that we'll eventually linked to instead of this content.